apollo-router/src/plugins/rhai/engine.rs

```diff
@@ -508,6 +523,12 @@ mod router_context {
         obj.with_mut(|response| response.context = context);
         Ok(())
     }
+    #[rhai_fn(get = "status_code", pure, return_raw)]
+    pub(crate) fn supergraph_first_response_status_code_get(
+        _obj: &mut SharedMut<supergraph::FirstResponse>,
+    ) -> Result<HeaderMap, Box<EvalAltResult>> {
+        Err(CANNOT_ACCESS_STATUS_CODE_ON_A_SUPERGRAPH_RESPONSE.into())
+    }
```

Comment:
I don't think this is the best way to control access to the status code on supergraph/execution. The best approach is to not implement the function.

So delete:
supergraph_first_response_status_code_get
execution_first_response_status_code_get
get_status_code_supergraph_deferred_response
get_status_code_execution_deferred_response

and update tests accordingly.

Reply:
Done ✅